### PR TITLE
[IE Myriad] Use instance of InferenceEngine::Core via ie::ICore interface in Myriad plugin

### PR DESCRIPTION
**Pull request by nikita-kud**
#-30632  
hddl PR 194

CI:  
- [ ] 1296

Nets:
- [ ] 147

Pre-commit [12992](https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/12992/) is passed

### CODE REVIEW

File: inference-engine/src/vpu/common/src/utils/ie_helpers.cpp [hunk 1]
```diff
@@ -175,4 +175,15 @@ void printTo(DotLabel& lbl, const ie::CNNLayerPtr& ieLayer) {
     subLbl.appendPair("precision", ieLayer->precision.name());
 }
 
+static ie::ICore* _core;
```
Comment (ilya-lavrenov) [inline inference-engine/src/vpu/common/src/utils/ie_helpers.cpp:178]: Am I right that it's thread-safe and work even if several cores are created?

Comment (nikita-kud) [reply inference-engine/src/vpu/common/src/utils/ie_helpers.cpp:178]: As I understand it, this is not related to multithreading. Each Core loads one "Myriad" plugin and `SetCore` method is called only once per instance of Core and plugin.

**Comment by DariaMityagina** [comment]
Approved the changes, they are fine. But the failures have to be fixed, as I can see out of logs there are some problems with adding new parameter like `candidate expects 2 arguments, 1 provided` and so on.

File: inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2020 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: Mocked core will be linked from `ieUnitTestUtils`. Is'n it? (`unit_test_utils\mocks\....hpp`)

**Comment by gladilov-gleb** [comment]
@Maxim-Doronin FYI

File: inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h (hunk 2 above)
Comment (nikita-kud) [reply inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: @iefode But this is deprecated tests..

Comment (nikita-kud) [reply inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: Do you mean, I need to link `unit_test_utils` from `tests` and `deprecated_tests`?

Comment (ilya-lavrenov) [reply inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: You need to put mock for core to unit_test_utils and link here.

Comment (iefode) [reply inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: `Unit_tests_utils` are linked to the deprecated unit tests...
image

Comment (nikita-kud) [reply inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h:1]: > You need to put mock for core to unit_test_utils  

It's already here. I need only remove `inference-engine/tests_deprecated/unit/engines/vpu/myriad_tests/helpers/mock_icore.h`

> Unit_tests_utils are linked to the deprecated unit test  

Ok, I get it. Thank you for the screenshot)

File: inference-engine/tests_deprecated/helpers/CMakeLists.txt [hunk 3]
```diff
@@ -37,11 +37,11 @@ endfunction()
 
 add_helpers(${TARGET_NAME})
 
-target_link_libraries(${TARGET_NAME} PUBLIC commonTestUtils)
+target_link_libraries(${TARGET_NAME} PUBLIC commonTestUtils gmock gtest)
```
Comment (iefode) [inline inference-engine/tests_deprecated/helpers/CMakeLists.txt:40]: `gmock` and `gtest` is necessary here?

Comment (nikita-kud) [reply inference-engine/tests_deprecated/helpers/CMakeLists.txt]: @iefode `gmock` only

Comment (iefode) [reply inference-engine/tests_deprecated/helpers/CMakeLists.txt]: `gmock` is linked to the `unit_test_utils`. I think it is extra for `ie_helpers`

File: inference-engine/src/vpu/myriad_plugin/myriad_executable_network.h [hunk 4]
```diff
@@ -35,17 +35,20 @@ class ExecutableNetwork : public InferenceEngine::ExecutableNetworkThreadSafeDef
...
-                               const MyriadConfig& config);
+                               const MyriadConfig& config,
+                               const ie::ICore* core);
 
     explicit ExecutableNetwork(std::istream& strm,
                                std::shared_ptr<IMvnc> mvnc,
                                std::vector<DevicePtr> &devicePool,
-                               const MyriadConfig& config);
+                               const MyriadConfig& config,
+                               const ie::ICore* core);
```
Comment (ilya-lavrenov) [inline inference-engine/src/vpu/myriad_plugin/myriad_executable_network.h:45]: `ExecutableNetwork` has a pointer to a plugin `_plugin`, you can extract `ICore` interface via `_plugin->GetCore()` inside the executable network.

File: inference-engine/tests_deprecated/helpers/mock_icore.hpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <gmock/gmock.h>
+#include "ie_icore.hpp"
+
+class MockICore : public InferenceEngine::ICore {
```
Comment (ilya-lavrenov) [inline inference-engine/tests_deprecated/helpers/mock_icore.hpp:10]: @iefode do we need to double `MockICore` inside the deprecated tests? I supposed we can re-use from `inference-engine/tests/ie_test_utils/unit_test_utils/mocks/cpp_interfaces/interface/mock_icore.hpp`

Comment (iefode) [reply inference-engine/tests_deprecated/helpers/mock_icore.hpp]: I agree with your point, we can reuse it

**Comment by Maxim-Doronin** [comment]
@nikita-kud, the mentioned in PR description validation build is not ok. Could you re-run validation?

**Comment by nikita-kud** [comment]
> Could you re-run validation?  

Unfortunately, pre-commit is still red:( So I will run the internal CI when it is fixed.

**Comment by nikita-kud** [comment]
@iefode @ilya-lavrenov please take a look.

**Comment by nikita-kud** [comment]
@Maxim-Doronin please take a look.
[Pre-commit](https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/12941/) passed

**Pull request merged**